models/world_coordinates.py

```diff
@@ -102,25 +103,56 @@ def get_upstreams(self, session=None):
         with SmartSession(session) as session:
             return session.scalars(sa.select(SourceList).where(SourceList.id == self.sources_id)).all()
         
-    def get_downstreams(self, session=None):
-        """Get the downstreams of this WorldCoordinates"""
-        # get the ZeroPoint that uses the same SourceList as this WCS
+    def get_downstreams(self, session=None, siblings=False):
+        """Get the downstreams of this WorldCoordinates.
+
+        If siblings=True  then also include the SourceLists, PSFs, ZPs and background objects
+        that were created at the same time as this WorldCoordinates.
+        """
+        from models.source_list import SourceList
+        from models.psf import PSF
         from models.zero_point import ZeroPoint
-        from models.image import Image
         from models.provenance import Provenance
-        with SmartSession(session) as session:
-            zps = session.scalars(sa.select(ZeroPoint) 
-                                  .where(ZeroPoint.provenance 
-                                         .has(Provenance.upstreams 
-                                              .any(Provenance.id == self.provenance.id)))).all()
-
-            subs = session.scalars(sa.select(Image)
-                                   .where(Image.provenance
-                                          .has(Provenance.upstreams
-                                               .any(Provenance.id == self.provenance.id)))).all()
-
-        downstreams = zps + subs
-        return downstreams
+
+        with (SmartSession(session) as session):
+            subs = session.scalars(
+                sa.select(Image).where(
+                    Image.provenance.has(Provenance.upstreams.any(Provenance.id == self.provenance.id))
+                )
+            ).all()
+            output = subs
```

Comment:
Same question here that I had in source_list -- won't this pull out all images whose provenances includes this world coordiantes' provenance, not just the ones that actually come from this wcs?